software-engineering-policies/TechnicalDebt/TechnicalDebtPolicy.md

```diff
@@ -1,48 +1,172 @@
+---
 
-## Additional Reading
+## Raising Technical Debt
 
-​<https://martinfowler.com/bliki/TechnicalDebt.html>
-<https://martinfowler.com/bliki/EstimatedInterest.html>
-<https://martinfowler.com/bliki/TechnicalDebtQuadrant.html>
+Technical debt identified in the current sprint but not being resolved as part of present work **MUST** be raised as a PBI in the relevant area of Azure DevOps. The relevant location is based on whether the technical debt is related to a product or a team.
+
+Ideally even if technical debt is going to be resolved immediately raising it is good for understanding its impact on UKHO work and identifying that technical debt was discovered.
+
+When raising technical debt, please add the following information to the Technical Debt PBI:
+
+### Title
+
+```Technical Debt - {Concise outline of Technical Debt item}```
+
+### Description
+
+The description of the Technical Debt item must include the following as a minimum:
+
+* The repository the Technical Debt lives within (a link to the repo would be good but the name would suffice)
+* Description of the issue in as much detail as you can provide, including class/file names if this can be included
+* A proposed solution or ideas for a fix **
+* Cost/Benefit and risk of the Technical Debt item, simply as a "why should this be fixed?/How long would it take?"
+
+### Tagging Strategy
+
+* The Technical Debt PBI **MUST** be tagged with the `````Technical Debt````` tag and should have any further appropriate tags, such as the following:
+  * Product Name (could be multiple, see the product list in AppRegister; [AppRegister](http://appregister/))
+  * Tech Debt Type:
+    * Security
+    * Knowledge
+    * Automation
+    * Testing
+    * Architecture
+    * Code
+    * Infrastructure
+  * Tech Debt Impact:
+    * Availability
+    * Portability
+    * Compatibility
+    * Performance
+    * Maintainability
+    * Extensibility  
+  * Out-Of-Support**
+  * Investigate To Remove** - This tag has been used to mark a Technical Debt item that may already have been completed or is going to be irrelevant due to other work.
+  * UKHO TD priority tag:
+    * TD1 - High business/technical value, high risk debt that needs to be paid off ASAP (a Security tag should automatically be considered for a TD1 prioritisation).
+    * TD2 - High business/technical value (cost reduction, blocker removing, maintainability improvement), lower risk, a change that should be worked on when time/opportunity permits and is not something that can be accepted or supported long term.
+    * TD3 - Tech debt that has been accepted as a risk but through paying off would add value through improving usability, maintainability, reliability or performance.
+    * TD4 - Accepted risk from the business, safe to leave until service reaches end of life. Worth tracking in case developers are working in the area and can complete as quick wins.
+
+** - Denotes where the tag is appropriate
+
+### Tech Debt Metric Strategy
```

Comment:
I think this section is a bit too much for right now, getting things Tagged with a level, type and effect I think will be  more value.

Could the Tech Debt Metric Strategy section be taken out of this PR and brought in separately with some more engagement with the teams.

Reply:
This PR is to provide the information for Principles to consider going forward, If you want to build a PR using this content as the next iteration of the policy then I would begin a new branch and take content across. 

I would say that in my research, I found that businesses who did not include these metric points made very little progress in their accuracy of TD tracking and thus did not make effective gains in it's management. I would champion this section as the most valuable part of this PR. While dev teams and individual engineers might not see the value here, the value to wider business planning and oversight is much greater.

Reply:
That is fine, I think the other bits are value we need now, and then this bit is in the future but not to be forgotten about future, maybe we can setup some workshops so people can get used to the options as it becomes a section in the policy document.

Reply:
The other bits are already in place in the existing policy so that's all good. Happy to run workshops on the other sections if the principles deem it as our chosen approach as I have been told to hand this off to them.